mfr/extensions/image/export.py

```diff
@@ -40,16 +41,19 @@ def export(self):
                 ratio = min(max_size[0] / image.size[0], max_size[1] / image.size[1])
                 self.metrics.add('ratio', ratio)
                 if ratio < 1:
-                    image = image.resize((round(image.size[0] * ratio), round(image.size[1] * ratio)), Image.ANTIALIAS)
-            if type in ['jpeg', 'jpg']:
-                # handle transparency
-                image = image.convert('RGBA')
-                exported_image = Image.new("RGBA", image.size, (255, 255, 255))
-                exported_image.paste(image, image)
-            else:
-                exported_image = image
-            exported_image.save(self.output_file_path, type)
-            exported_image.close()
+                    image = image.resize((round(image.size[0] * ratio),
+                                round(image.size[1] * ratio)), Image.ANTIALIAS)
+            # handle transparency
+            if image.mode in ('RGBA', 'LA') and type in ['jpeg', 'jpg']:
+                # JPEG has no transparency, so anything that was transparent gets changed to
```

Comment:
any other possible modes? Also add a link to the pillow issue you used to solve this.

Reply:
complete